src/components/forms/PlanForm/components/retirePlanForm.tsx

```diff
@@ -0,0 +1,215 @@
+import { getUser } from '@onaio/session-reducer';
+import { Field, Form, Formik } from 'formik';
+import moment from 'moment';
+import React from 'react';
+import { Button, Col, FormGroup, Label, Row } from 'reactstrap';
+import { ACTION_UUID_NAMESPACE } from '../../../../configs/env';
+import {
+  ACTIVITY_DONE_BEFORE,
+  AN_ERROR_OCCURRED,
+  ASSIGNED_WRONG_PLAN,
+  CANCEL,
+  CANCELED_PLAN,
+  DUPLICATED_PLAN,
+  ENTERED_WRONG_ADDRESS,
+  OTHER_REASON,
+  OTHER_REASON_LABEL,
+  PROCEED,
+  RETIRE_PLAN_REASON,
+  SAME_SUB_VILLAGE,
+} from '../../../../configs/lang';
+import { PlanDefinition } from '../../../../configs/settings';
+import {
+  EVENT_LABEL,
+  FORM_SUBMISSION_FIELD,
+  OPENSRP_EVENT_ENDPOINT,
+  PLAN_DEFINITION,
+  RETIRE_PLAN,
+  RETIRE_REASON,
+} from '../../../../constants';
+import { displayError } from '../../../../helpers/errors';
+import { generateNameSpacedUUID } from '../../../../helpers/utils';
+import { OpenSRPService } from '../../../../services/opensrp';
+import store from '../../../../store';
+
+/** Formik set submitting type  */
+export type SetSubmittingType = (isSubmitting: boolean) => void;
+
+/** Retire plan from props interface */
+interface RetirePlanFormProps {
+  cancelCallBack: () => void;
+  payload: PlanDefinition;
+  setSubmittingCallBack: SetSubmittingType;
+  savePlan: (payload: PlanDefinition, setSubmitting: SetSubmittingType) => void;
+}
+
+/** retiring Form values interface */
+interface RetiringFormValues {
+  otherReason: string;
+  retireReason: string;
+}
+
+/** Define reasons, for translation purposes */
+const otherReason: string = 'OTHER';
+export const RetireReasonOptions: { [key: string]: string } = {
+  ACTIVITY_DONE: ACTIVITY_DONE_BEFORE,
+  CANCELED: CANCELED_PLAN,
+  DUPLICATE: DUPLICATED_PLAN,
+  SAME_VILLAGE: SAME_SUB_VILLAGE,
+  WRONG_ADDRESS: ENTERED_WRONG_ADDRESS,
+  WRONG_PLAN: ASSIGNED_WRONG_PLAN,
+  // tslint:disable-next-line: object-literal-sort-keys
+  [otherReason]: OTHER_REASON,
```

Comment:
Why not have `OTHER` here instead of `otherReason`?

Reply:
`otherReason` is a constant assigned `OTHER`. This was to avoid having to type `OTHER` when checking if user has selected other reason